src/lib.rs

```diff
@@ -862,3 +868,146 @@ impl ConstantTimeLess for u32 {}
+}
+
+impl<T: ConstantTimeGreater + ConstantTimeLess + ConstantTimeEq> ConstantTimePartialOrd for T {
+    /// We do not assume a total ordering for `T`, so we have to individually check "less than",
+    /// "equal", and "greater". This also respects non-default implementations of `ct_lt()`.
+    fn ct_partial_cmp(&self, other: &Self) -> CtOption<Ordering> {
+        let is_eq = self.ct_eq(other);
+        let is_lt = self.ct_lt(other);
+        let is_gt = self.ct_gt(other);
+
+        static PARTIAL_ORDERS: [CtOption<Ordering>; 4] = [
+            CtOption {
+                value: Ordering::Equal,
+                is_some: Choice::of_bool(false),
+            },
+            CtOption {
+                value: Ordering::Equal,
+                is_some: Choice::of_bool(true),
+            },
+            CtOption {
+                value: Ordering::Less,
+                is_some: Choice::of_bool(true),
+            },
+            CtOption {
+                value: Ordering::Greater,
+                is_some: Choice::of_bool(true),
+            },
+        ];
+        *index_mutually_exclusive_logical_results(&PARTIAL_ORDERS, [is_eq, is_lt, is_gt])
+    }
+}
+
+/// An [`Ord`][core::cmp::Ord]-like trait for constant-time comparisons.
+///
+/// This trait can be automatically implemented for types supporting the "equals" and "greater"
+/// comparisons.
+///
+/// # Example
+///
+/// ```
+/// use std::cmp::Ordering;
+/// use subtle::ConstantTimeOrd;
+/// let x: u8 = 5;
+/// let y: u8 = 13;
+///
+/// assert_eq!(x.ct_cmp(&x), Ordering::Equal);
+/// assert_eq!(x.ct_cmp(&y), Ordering::Less);
+/// assert_eq!(y.ct_cmp(&x), Ordering::Greater);
+/// ```
+pub trait ConstantTimeOrd: ConstantTimeEq + ConstantTimeGreater {
+    /// This method returns an ordering between `self` and other`.
+    ///
+    /// Although this method should never need to be overridden, it is exposed as a default method
+    /// here to force types to explicitly implement this trait. This ensures that types which are
+    /// *only* partially orderable do not pick up an incorrect `ConstantTimeOrd` impl just by
+    /// implementing the pairwise comparison operations. Contrast this with
+    /// [`ConstantTimePartialOrd`], which is automatically implemented for all types implementing
+    /// [`ConstantTimeGreater`], [`ConstantTimeLess`], and [`ConstantTimeEq`].
+    ///
+    /// Here we assume a total ordering for `T`, so we need to check only "equal" and "greater", and
+    /// can assume "less" if both `ct_eq()` and `ct_gt()` are false.
+    ///
+    /// This method should execute in constant time.
+    fn ct_cmp(&self, other: &Self) -> Ordering {
+        let is_gt = self.ct_gt(other);
+        let is_eq = self.ct_eq(other);
+
+        static ORDERS: [Ordering; 3] = [Ordering::Less, Ordering::Greater, Ordering::Equal];
+        *index_mutually_exclusive_logical_results(&ORDERS, [is_gt, is_eq])
```

Comment:
It seems like you should be able to build this on top of `ConditionallySelectable`?

```suggestion
        let ret = Ordering::Less;
        let ret = Ordering::conditional_select(&ret, &Ordering::Equal, is_eq);
        Ordering::conditional_select(&ret, &Ordering::Greater, is_gt)
```

Granted that doesn't ensure mutual exclusivity, although that seems implemented easily enough via a `debug_assert!`

Reply:
Thank you so much--I had totally missed the purpose of the `ConditionallySelectable` trait! Done in https://github.com/dalek-cryptography/subtle/pull/98/commits/ff01955eadc3f65979e86d00432801a70932f8a4! Will also be updating #99 to use `ConditionallySelectable` like this as well.